Application/EdFi.Ods.Tests/EdFi.Ods.Security/Authorization/EntityAuthorization/EntityInstanceViewBasedFilterAuthorizerTests.cs

```diff
@@ -0,0 +1,571 @@
+                            // ClaimParameterName = "ClaimParam",
+                            // ClaimEndpointValues = [123]
+                        },
+                        FilterDefinition = new ViewBasedAuthorizationFilterDefinition(
+                            "filter-name",
+                            "CustomTestView",
+                            "view-source-endpoint",
+                            "view-target-endpoint",
+                            null, null, null)
+                    }
+                }
+            }
+        };
+
+        var claimEducationOrganizationIds = new List<long> { 123 };
+
+        A.CallTo(() => _sqlBuilder.BuildExistenceCheckSql(resultsWithPendingExistenceChecks)).Returns("Generated SQL");
+
+        A.CallTo(
+                () => _queryExecutor.ExecuteAsync(
+                    "Generated SQL",
+                    A<KeyValuePair<string, object>[]>.Ignored,
+                    resultsWithPendingExistenceChecks,
+                    claimEducationOrganizationIds))
+            .Returns(Task.FromResult<int?>(0));
+
+        A.CallTo(() => _hintProviders[0].GetFailureHint("CustomTestView")).Returns("Custom test hint");
+        A.CallTo(() => _hintProviders[0].GetFailureHint("TestView")).Returns("Test hint");
+
+        // Act & Assert
+        var exception = Should.ThrowAsync<SecurityAuthorizationException>(
+            () => _authorizer.PerformViewBasedAuthorizationAsync(
+                resultsWithPendingExistenceChecks,
+                claimEducationOrganizationIds,
+                AuthorizationPhase.ExistingData));
+
+        exception.Result.Detail.ShouldStartWith("Access to the requested data could not be authorized.");
+        exception.Result.Detail.ShouldContain("Hint: Custom test hint");
+        (exception.Result as IEdFiProblemDetails).Errors.ShouldContain("The caller is not authorized to perform the requested operation on the item based on the existing value of the 'TestParam' property of the item.");
+    }
+
+    [Test]
+    public void PerformViewBasedAuthorizationAsync_ShouldThrowSecurityAuthorizationException_WithoutHints_WhenNoHintsAvailable_ForCustomViewBasedAuthorization()
+    {
+        // Arrange
+        var resultsWithPendingExistenceChecks = new[]
+        {
+            new AuthorizationStrategyFilterResults
+            {
+                FilterResults = new[]
+                {
+                    new FilterAuthorizationResult
+                    {
+                        FilterContext = new AuthorizationFilterContext
+                        {
+                            SubjectEndpointNames = new[] { "TestParam" },
+                            SubjectEndpointValues = new object[] { 456 },
+                            // Custom view-based authorization will have no claim values
+                            // ClaimParameterName = "ClaimParam",
+                            // ClaimEndpointValues = [123]
```

Comment:
Can this commented out code be removed?

Reply:
I've removed the two lines of "code" but kept the comment as it is pertinent to Custom view-based authorization testing for someone looking at the tests.